submit_and_compare/submit_and_compare.py

```diff
@@ -147,8 +196,20 @@ def student_submit(self, submissions, suffix=''):
         '''
         Save student answer
         '''
+        if self.max_attempts > 0 and self.count_attempts >= self.max_attempts:
```

Comment:
to make this more easily understandable you could define:
```python
unlimited_attempts = self.max_attempts < 0
```

and then the check here could be:
```python
if not unlimited_attempts and self.count_attempts >= self.max_attempts:
```